templates/misc/icon.twig

```diff
@@ -1 +1,7 @@
-<span class="hel-icon hel-icon--{{ icon }} {{ class }}" {% if label %} aria-label="{{ label }}" {% else %} aria-hidden="true" {% endif %}></span>
+{% set icon_labelled_by = "hdbt-icon--" ~ random() %}
+
+<span class="hel-icon hel-icon--{{ icon }} {{ class }}" role="img" {% if label %} aria-labelledby="{{ icon_labelled_by }}" {% else %} aria-hidden="true" {% endif %}>
+  {% if label %}
+    <span class="visually-hidden" id="{{ icon_labelled_by }}">{{ label }}</span>
```

Comment:
Should this be `is-hidden` instead as `aria-labelledby` should work just as well with `display:none`?

Reply:
This is switched now.